include/dav1d/picture.rs

```diff
@@ -292,11 +295,14 @@ impl Rav1dPictureDataComponent {
         self.as_strided_mut_ptr::<BD>().cast_const()
     }
 
-    fn as_dav1d(&self) -> Option<NonNull<c_void>> {
+    fn as_dav1d(&self) -> Option<Unique<c_void>> {
         if self.byte_len() == 0 {
             None
         } else {
-            NonNull::new(self.as_strided_byte_mut_ptr().cast())
+            let ptr = NonNull::new(self.as_strided_byte_mut_ptr()).unwrap();
+            // SAFETY: The `ptr` originally comes from a `Unique` in `Rav1dPictureDataComponentInner::ptr`.
+            let ptr = unsafe { Unique::new(ptr) };
```

Comment:
Again, how is this unique? Can this method return a NonNull instead?

Reply:
Because `self.as_strided_byte_mut_ptr()` is derived from `Rav1dPictureDataComponentInner::ptr`, which is a `Unique`.  I can say that more explicitly if that's clearer.

Reply:
e.g. can I just call this twice and get two `Unique`s? Doesn't that violate the invariants?

Reply:
Ugh, yeah, you're right. 

Why does `core::ptr::Unique: Copy + Clone` then?

Reply:
No idea.